src/js/core/common.js

```diff
@@ -143,19 +143,42 @@ export class StopWatch {
         console.log(currentName, '[' + this.getElapsedMilliseconds() + 'ms]', '[' + this.getElapsedSeconds() + 's]');
     }
 }
-
 export function subscribeMixin (target) {
     var listeners = new Set();
 
     return Object.assign(target, {
 
-        subscribe(listener) {
+        on (type, f) {
+            let listener = {};
+            listener[type] = f;
             listeners.add(listener);
         },
 
-        on(type, f) {
-            let listener = {};
-            listener[type] = f;
+        off (type, f) {
+            if (f) {
+                let listener = {};
+                listener[type] = f;
+                listeners.delete(listener);
+            }
+            else {
```

Comment:
Can you document what subscribeMixin is supposed to do? I should have asked earlier but never did...
We should also figure out what you're trying to accomplish and try to do it in a way that doesn't need as many nested loops / ifs